docs/developer-guide.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Developer Guide
+toc: true
+weight: 5010
+---
+# Developer Guide
+Welcome to the Crossplane Developer Guide!
+
+## Overview 
+Infra Stacks like
+[stack-gcp][stack-gcp], [stack-aws][stack-aws], and [stack-azure][stack-azure] extend Crossplane
+to support managed service provisioning (databases, caches, buckets), secure
+connectivity (VPCs, subnets, peering, ACLs, secrets), and provisioning managed
+Kubernetes clusters on demand to further isolate the blast radius of
+applications. 
+
+Infra Stacks are typically pre-built and published to the [Stacks
+registry][stack-registry], where they can
+be installed by a cluster administrator using a
+[`ClusterStackInstall`][stack-install-docs]
+kind via the Kubernetes API or with the [`stack
+install`][crossplane-cli-usage] command.
+
+App Stacks depend on Infra Stacks like
+[stack-gcp][stack-gcp], [stack-aws][stack-aws], or [stack-azure][stack-azure]
+ to provide the managed services they depend on via the Kubernetes API.
+
+App Stacks may also be pre-built and published to the [Stacks
+registry][stack-registry] where they can
+be deployed by application teams using a
+[`StackInstall`][crossplane-cli-usage]]
+kind via the Kubernetes API or with the [`stack
+install`][crossplane-cli-usage]) command.
+
+## Infra Stacks 
+### Using Infra Stacks 
```

Comment:
I might consider putting the usage of stacks in a section at the top of the document; the section might look something like:

```
## Goals

Here's what we'll be learning more about:
* Stuff

Here's what we will **not** be learning about:
* Using Stacks. Read more about how to use them in the [stack guides][link-thing]
```

Anyway, that is what I am thinking based on the document sounding like it is a general guide for developing Crossplane.